test/mocks/test-data/api-responses.js

```diff
@@ -0,0 +1,422 @@
+// Mock transformation codes for different test scenarios
+const transformationMocks = {
+  // Simple JS transformation
+  'simple-js-transform': {
+    versionId: 'simple-js-transform',
+    code: `
+      export function transformEvent(event, metadata) {
+        event.properties = event.properties || {};
+        event.properties.transformed = true;
+        event.properties.transformationType = 'simple';
+        return event;
+      }
+    `,
+    name: 'Simple Transform',
+    codeVersion: '1',
+    language: 'javascript',
+    workspaceId: 'test-workspace-1',
+    imports: []
+  },
+
+  // Async JS transformation
+  'async-js-transform': {
+    versionId: 'async-js-transform',
+    code: `
+      export async function transformEvent(event, metadata) {
+        // Simulate async operation without setTimeout (not available in VM)
+        await Promise.resolve();
+        
+        event.properties = event.properties || {};
+        event.properties.transformed = true;
+        event.properties.transformationType = 'async';
+        event.properties.processedAt = new Date().toISOString();
+        return event;
+      }
+    `,
+    name: 'Async Transform',
+    codeVersion: '1',
+    language: 'javascript',
+    workspaceId: 'test-workspace-1',
+    imports: []
+  },
+
+  // JS transformation with fetch calls
+  'fetch-js-transform': {
```

Comment:
didn't get this. shouldn't this call fetch? how different is this from `simple-js-transform`?